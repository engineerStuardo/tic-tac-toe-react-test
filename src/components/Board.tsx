import Square from './Square';

import { BoardProps } from './interfaces';

const Board = ({ squares }: BoardProps) => (
  <div data-testid='board' className='board'>
    {squares.map((square, i) => (
      <Square key={i} value={square} />
    ))}
  </div>
);

export default Board;
