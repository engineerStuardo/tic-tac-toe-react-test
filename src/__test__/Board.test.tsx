import React from 'react';
import { render, screen } from '@testing-library/react';

import Board from '../components/Board';

const squares = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const onClick = jest.fn();

describe('Board component', () => {
  test('Should match a snapshot', () => {
    const { container } = render(<Board squares={squares} onClick={onClick} />);
    expect(container).toMatchSnapshot();
  });

  test('Should be a "board" class in the component', () => {
    render(<Board squares={squares} onClick={onClick} />);
    expect(screen.getByTestId('board')).toBeTruthy();
  });
});
