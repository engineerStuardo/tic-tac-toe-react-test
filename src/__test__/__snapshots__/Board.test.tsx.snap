// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component Should match a snapshot 1`] = `
<div>
  <div
    class="board"
    data-testid="board"
  >
    <button
      class="squares 1"
    >
      1
    </button>
    <button
      class="squares 2"
    >
      2
    </button>
    <button
      class="squares 3"
    >
      3
    </button>
    <button
      class="squares 4"
    >
      4
    </button>
    <button
      class="squares 5"
    >
      5
    </button>
    <button
      class="squares 6"
    >
      6
    </button>
    <button
      class="squares 7"
    >
      7
    </button>
    <button
      class="squares 8"
    >
      8
    </button>
    <button
      class="squares 9"
    >
      9
    </button>
  </div>
</div>
`;
